code/modules/mob/living/carbon/human/species/necromorph/twitcher.dm

```diff
@@ -170,6 +170,9 @@ All of these properties combined make Step Strike tricky and disorienting to use
 			var/vector2/delta = get_new_vector(A.x - x, A.y - y)
 			delta = delta.ToMagnitude(3)
 			var/turf/blink_target = locate(x+delta.x, y+delta.y, z)
+			if(!check_trajectory(blink_target, src, pass_flags))
```

Comment:
if something blocks then the trajectory should be scaled back to a smaller magnitude until it doesn't collide, run a loop for that reducing it by one each time